tests/functional/conftest.py

```diff
@@ -0,0 +1,43 @@
+import subprocess
+
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def install_dcgm_snap():
+    """Install the snap and enable openstack-exporter service for testing."""
+    snap_build_name = "charmed-openstack-exporter_*.snap"
+    config_dir_path = "/etc/openstack"
+    config_file_path = f"{config_dir_path}/clouds.yaml"
+    config_test_file_path = f"{config_dir_path}/test.yaml"
+
+    subprocess.check_call(
+        f"sudo snap install --dangerous {snap_build_name}",
+        shell=True,
+    )
+
+    # Create the directory if it doesn't exist and create the file
+    subprocess.check_call(["sudo", "mkdir", "-p", config_dir_path])
+    subprocess.check_call(["sudo", "touch", config_file_path])
+    subprocess.check_call(["sudo", "touch", config_test_file_path])
+
+    # To allow the snap to read from the /etc/openstack/, manually add the connection:
+    subprocess.check_call(["sudo", "snap", "connect", "charmed-openstack-exporter:etc-openstack"])
+    subprocess.check_call(
+        [
+            "sudo",
+            "snap",
+            "set",
+            "charmed-openstack-exporter",
+            f"os-client-config={config_file_path}",
+            "cloud=cloud",
+        ]
+    )
+    subprocess.check_call(["sudo", "snap", "start", "charmed-openstack-exporter"])
+
+    yield
+
+    subprocess.check_call(["sudo", "snap", "remove", "--purge", "charmed-openstack-exporter"])
+    subprocess.check_call(["sudo", "rm", config_file_path])
+    subprocess.check_call(["sudo", "rm", config_test_file_path])
+    subprocess.check_call(["sudo", "rmdir", config_dir_path])
```

Comment:
Because of all these destructive operations in the func tests, could you add a warning to the readme and tox.ini target (and anywhere else appropriate?) about running it locally on your workstation?

Reply:
Added a warning to the README file

Reply:
could we store the config file in TMPDIR instead? It'd still be nice to clean up after ourselves, but risks would be minimized

Reply:
https://github.com/canonical/charmed-openstack-exporter-snap/blob/6096b2fda9b29ad6285a2f7bc0c387c80f59efb8/snap/snapcraft.yaml#L162

When I saw that, I realized we could do the same for the dcgm-snap, avoiding the need for the user to put the files in the snap's folder - speaking about UX. But I didn't know that we can do this :(

Reply:
Don't worry too much - putting files in the snap folder is preferrable if possible. The snap needs a separate review process for any system-files interface too, and would probably only be approved if it's strictly necessary. :)